docs/content/manual/v1.7/manual.yml

```diff
@@ -3017,10 +3017,16 @@ sections:
 
           The `nth(n)` function extracts the nth value of any array at `.`.
 
+          `first` and `nth` output `empty` if the generator has insufficient
```

Comment:
I wonder if "produces no output" or "outputs no value" is better then "output empty"? seems to be what is used in other places

Reply:
I found the documentation for `try` also uses the phrase "output `empty`". In this case I think it's a bit more explicit than "outputs no value".

Reply:
Ok, i mostly want consistency but i think i personally prefer "no output" as mentioning empty feel a bit confusing, ex would we document the function `empty` as producing `empty`? hmm

Reply:
I have updated the docs to say "produce no values" instead

Reply:
👍 let's see what the other maintainers think

But i do wonder what the resigning was for `last` to return `null`?

Reply:
I assume it was just the easiest to implement. `last` is implemented as:
```jq
def last(g): reduce g as $item (null; $item);
```
Reduce can't really return `empty` fundamentally so you'd need to do something more complex to match the behavior of `first` and `nth`.

At this point it would probably break too many to change it so it's better to just document the current behavior.

Reply:
One potential "fixed" last that outputs no values for an empty generator would be this:

```jq
def last(g): reduce g as $item ([0,null]; [first+1,$item])|if first>0 then last else empty end;
```

A lot less elagent than the current implementation though.

Reply:
See #1869. This is a bug that should be fixed in the future.

Reply:
@itchyny 👍  oh long discussion, seems fascinatingly messy :)

Reply:
That branch is 5 years old, I doubt it will still be merged

Reply:
It's better to use byte code to avoid boxing like the PR even if it makes more messy.